### PR TITLE
Chapter 8

### PR DESCRIPTION
**Pull request by sebing**
SxG Chapter 08

### CODE REVIEW

File: chapter08/sxg/ex_8_2.pl [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!perl
+use warnings;
+use strict;
+
+my $input;
+
+if ( @ARGV == 1 ){
```
Comment (codavex) [inline chapter08/sxg/ex_8_2.pl:7]: You don't need to change the program to look at argv, you could just redirect a file into its STDIN

File: chapter08/sxg/ex_8_6.pl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#use utf8;
+
+my $marker = "☺";
+
+if ( @ARGV == 1 ){
+	while (<>) {
+		my $line = $_;
+	  	chomp($line);
+	  	if ($line =~ /\s\z/) {
+	    	print "$line$marker\n";
```
Comment (codavex) [inline chapter08/sxg/ex_8_6.pl:13]: ah, just realised I did this exercise wrong. :blush:

File: chapter08/sxg/ex_8_2.pl (hunk 1 above)
Comment (sebing) [inline chapter08/sxg/ex_8_2.pl:7]: Then I have to add, user input messages( i was lazy)

File: chapter08/sxg/ex_8_1.pl [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!perl
+use warnings;
+use strict;
+
+print "Enter a string: \n";
+
+while (<>) {                   # take one input line at a time
+	chomp;
+	if (/YOUR_PATTERN_GOES_HERE/) {
+		print "Matched: |$`<$&>$'|\n";  # the special match vars
```
Comment (pratyp) [inline chapter08/sxg/ex_8_1.pl:10]: think you forgot to replace <code>YOUR_PATTERN_GOES_HERE</code> :smile:

**Pull request merged**